gpcontrib/arenadata_toolkit/arenadata_toolkit--1.3--1.4.sql

```diff
@@ -0,0 +1,399 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.3--1.4.sql */
+
+DO $$
+BEGIN
+	IF EXISTS (SELECT 1
+			   FROM pg_tables
+			   WHERE schemaname = 'arenadata_toolkit' AND
+			         tablename = 'db_files_current')
+	THEN
+		-- Use "IF" into another "IF" because in other case "<table_name>::regclass"
+		-- leads to error if the such table does not exist.
+		IF NOT EXISTS (SELECT 1
+					   FROM pg_attribute
+					   WHERE attname = 'tablespace_location' AND
+					         NOT attisdropped AND
+					         attrelid = 'arenadata_toolkit.db_files_current'::regclass)
+		THEN
+			ALTER TABLE arenadata_toolkit.db_files_current
+			ADD COLUMN tablespace_location TEXT;
+		END IF;
+	END IF;
+END $$;
+
+DO $$
+BEGIN
+	IF EXISTS (SELECT 1
+			   FROM pg_tables
+			   WHERE schemaname = 'pg_temp' AND
+			         tablename = 'db_files_current')
+	THEN
+		DROP TABLE pg_temp.db_files_current;
+	END IF;
+END $$;
+
+DO $$
+BEGIN
+	IF EXISTS (SELECT 1
+			   FROM pg_views
+			   WHERE schemaname = 'arenadata_toolkit' AND
+			         viewname = '__db_files_current')
+	THEN
+		-- Use "IF" into another "IF" because in other case "<table_name>::regclass"
+		-- leads to error if the such table does not exist.
+		IF NOT EXISTS (SELECT 1
+					   FROM pg_attribute
+					   WHERE attname = 'tablespace_location' AND
+					         NOT attisdropped AND
+					         attrelid = 'arenadata_toolkit.__db_files_current'::regclass)
+		THEN
+			CREATE OR REPLACE VIEW arenadata_toolkit.__db_files_current
+			AS
+			SELECT
+				c.oid AS oid,
+				c.relname AS table_name,
+				n.nspname AS table_schema,
+				c.relkind AS type,
+				c.relstorage AS storage,
+				d.datname AS table_database,
+				t.spcname AS table_tablespace,
+				dbf.segindex AS content,
+				dbf.segment_preferred_role AS segment_preferred_role,
+				dbf.hostname AS hostname,
+				dbf.address AS address,
+				dbf.full_path AS file,
+				dbf.size AS file_size,
+				dbf.modified_dttm AS modifiedtime,
+				dbf.changed_dttm AS changedtime,
+				CASE
+					WHEN 'pg_default' = t.spcname THEN gpconf.datadir || '/base'
+					WHEN 'pg_global' = t.spcname THEN gpconf.datadir || '/global'
+					ELSE (SELECT tblspc_loc
+						  FROM gp_tablespace_segment_location(t.oid)
+						  WHERE gp_segment_id = dbf.segindex)
+					END AS tablespace_location
```

Comment:
may be move this column to arenadata_toolkit.__db_segment_files view, because of it already has join with gp_segment_configuration?